bundles/org.openhab.binding.pushbullet/src/main/java/org/openhab/binding/pushbullet/internal/handler/PushbulletHandler.java

```diff
@@ -273,21 +271,10 @@ private boolean sendPush(PushRequest request) {
      * Helper method to get the file name for a given content
      *
      * @param content the file content
-     * @return the file name if available, otherwise null
+     * @return the file name
      */
-    private @Nullable String getContentFileName(String content) {
-        if (content.startsWith("data:")) {
-            return IMAGE_FILE_NAME;
-        }
-        try {
-            Path fileName = Path.of(content.startsWith("http") ? new URL(content).getPath() : content).getFileName();
-            if (fileName != null) {
-                return fileName.toString();
-            }
-        } catch (MalformedURLException e) {
-            logger.debug("Malformed url content: {}", e.getMessage());
-        }
-        return null;
+    private String getContentFileName(String content) {
+        return content.startsWith("data:") ? IMAGE_FILE_NAME : Path.of(content).getFileName().toString();
```

Comment:
It seems you removed URL support, is it because it's unused by the relevant Thing actions?

Reply:
`Path.of` is able to parse URL but I overlooked the fact that an URL with query string would not. Anyway, I just replaced the URL constructor with `URI.create` in the end.